src/main/java/org/sciborgs1155/robot/drive/SparkModule.java

```diff
@@ -135,9 +157,63 @@ public void resetEncoders() {
     driveEncoder.setPosition(0);
   }
 
+  @Override
+  public SwerveModuleState state() {
+    return new SwerveModuleState(driveVelocity(), rotation());
+  }
+
+  @Override
+  public SwerveModulePosition position() {
+    return new SwerveModulePosition(drivePosition(), rotation());
+  }
+
+  @Override
+  public SwerveModuleState desiredState() {
+    return setpoint;
+  }
+
+  @Override
+  public void setDriveSetpoint(double velocity) {
+    drivePID.setReference(velocity, ControlType.kVelocity);
```

Comment:
No feedforward is used in closed-loop control currently. It can be added using the overloaded method
```suggestion
    public REVLibError setReference(
      double value, CANSparkBase.ControlType ctrl, int pidSlot, double arbFeedforward)
```
with the feedforward value being calculated with WPILIB's class.